server/src/main/java/io/druid/server/QueryResource.java

```diff
@@ -151,6 +152,7 @@ public Response doPost(
         );
       }
 
+      Thread.currentThread().setName("ProcessQuery-"+queryId);
```

Comment:
Also I'm not sure this method blocks. I thought it blocked in the response's StreamingOutput (but haven't profiled this code area in a very long time)